### PR TITLE
Add confirmation checkbox to finish page

### PR DESCRIPTION
**Pull request by shanear**
Sticking to the style guide for now so it won't match the mocks perfectly. We can address the styleguide later.

connects #3437

### CODE REVIEW

File: client/app/containers/EstablishClaimPage/EstablishClaimNote.jsx [hunk 1]
```diff
@@ -154,7 +154,6 @@ export class EstablishClaimNote extends BaseForm {
       <div className="route-claim-confirmNote-wrapper">
         <Checkbox
           label="I confirm that I have created a VBMS note to help route this claim."
-          fullWidth={true}
```
Comment (shanear) [inline client/app/containers/EstablishClaimPage/EstablishClaimNote.jsx]: turns out this also does nothing :neutral_face:

Comment (nickheiner-usds) [reply client/app/containers/EstablishClaimPage/EstablishClaimNote.jsx]: Thanks for figuring that out. Glad that we are learning more as part of this process.

**Comment by NickHeiner** [comment]
This does not match the following AC:

> - A statement with a checkbox displays below the issues on appeal, and the "I've completed all steps" button is disabled until this checkbox is selected. It is unselected by default.

Is that out of scope for this PR?

File: client/app/intake/pages/finish.jsx [hunk 2]
```diff
@@ -33,6 +38,21 @@ class Finish extends React.PureComponent {
...
+        label={
+          <span>
+            I confirm that I have completed all of the steps above.
+            I understand that selecting the
+            <b> { submitText } </b>
+            button below will close the VACOLS record.
+          </span>
+        }
+        name="confirm-finish"
+        required={true}
```
Comment (NickHeiner) [inline client/app/intake/pages/finish.jsx]: No need for the `={true}`. Just do:

```
required
```

**Comment by NickHeiner** [comment]
This looks good aside from not matching what the ticket asks for. :smile: The ticket calls for the "next" button to be disabled, instead of having an error state if the "next" button is clicked before the checkbox is clicked. Which behavior is correct? Did we change the goal without updating the AC?

**Comment by shanear** [comment]
I asked abby and we are going with the guidance on the styleguide:
screen shot 2017-10-18 at 4 48 46 pm
I will note that in the AC

File: client/app/intake/redux/reducer.js [hunk 3]
```diff
@@ -205,6 +207,22 @@ export const reducer = (state = mapDataToInitialState(), action) => {
...
+        finishConfirmed: {
+          $set: action.payload.isConfirmed
+        }
+      }
+    });
+  case ACTIONS.COMPLETE_INTAKE_NOT_CONFIRMED:
+    return update(state, {
+      rampElection: {
+        finishConfirmedError: {
+          $set: "You must confirm you've completed the steps"
```
Comment (NickHeiner) [inline client/app/intake/redux/reducer.js]: I don't see this copy in a spec anywhere but I'll assume that @abbyraskinUSDS is ok with it.

Comment (shanear) [reply client/app/intake/redux/reducer.js]: Yeah, not worth the bother.

**Pull request merged**